### PR TITLE
feat/ post update with image

### PR DESCRIPTION
**Pull request by mikco-hipe**
### OVERVIEW

Added feature on Post Edit with image
UI is not yet final

### SNIPPETS
Screenshot 2023-10-13 152805
Screenshot 2023-10-13 152833
Screenshot 2023-10-13 152942
Screenshot 2023-10-13 152858

### CODE REVIEW

File: laravel-app/app/Services/PostService.php [hunk 1]
```diff
@@ -63,4 +64,35 @@ public function sharePost(UserPost $post): UserPost
...
+                        'file_path' => $imagePath,
+                    ],
+                );
+            }
+
+            return true;
+        } catch (\Exception $e) {
+            return false;
+        }
+    }
```
Comment (hipe01mark) [inline laravel-app/app/Services/PostService.php:97]: Prevent using else in return statement.

https://medium.com/@andrerferrer/unnecessary-else-after-return-no-else-return-43e697575dc1#:~:text=So%2C%20what%20does%20it%20mean,NEED%20the%20%60else%60%20statement.&text=This%20is%20a%20case%20in,if%20the%20number%20is%20even.

Comment (mikco-hipe) [reply laravel-app/app/Services/PostService.php:97]: yes sir 🙇

File: laravel-app/app/Services/PostService.php [hunk 2]
```diff
@@ -63,4 +64,30 @@ public function sharePost(UserPost $post): UserPost
 
         return $sharedPost;
     }
+
+    /**
+     * To update the image on post
+     */
+    public function updatePostImage(UserPost $post, UploadedFile $image): bool
```
Comment (bpocrafael) [inline laravel-app/app/Services/PostService.php:71]: can you make this function descriptive as that it is return a boolean

Comment (bpocrafael) [reply laravel-app/app/Services/PostService.php:71]: also i was thinking is this for updating image? then maybe void?

Comment (bpocrafael) [reply laravel-app/app/Services/PostService.php:71]: oh, if its bool why cant you make it more descriptive as what it should be. the function name is updatePostImage

Comment (mikco-hipe) [reply laravel-app/app/Services/PostService.php:71]: Yes sir, I saw a stakoverflow thread about the naming convention on function that returns **bool**
https://stackoverflow.com/questions/1370840/naming-conventions-what-to-name-a-method-that-returns-a-boolean

I decided to rename the function `isPostImageUpdatable()`

Here is my call function on my `PostController.php`
 `$updatedImage = $this->postService->isPostImageUpdatable($post, $image);`

It is much more readable on what does the function returns.

File: laravel-app/app/Http/Controllers/PostController.php [hunk 3]
```diff
@@ -88,14 +88,22 @@ public function update(UserPostRequest $request, UserPost $post): RedirectRespon
...
 
-            return redirect()->route('post.show', ['post' => $post])->with($success);
+        if ($image instanceof UploadedFile) {
+            $updatedImage = $this->postService->isPostImageUpdatable($post, $image);
+            $successMessage = $updatedImage ? 'Post and image updated successfully' : 'Image updated successfully';
+        } elseif ($updated) {
+            $successMessage = 'Post updated successfully';
+        } else {
+            return redirect()->back()->with('error', 'Failed to update post and/or image');
         }
```
Comment (hipe01mark) [inline laravel-app/app/Http/Controllers/PostController.php:104]: You can prevent using else in the if statement. Try to research more

Comment (mikco-hipe) [reply laravel-app/app/Http/Controllers/PostController.php:104]: Yes sir 🙇

**Review by hipe01mark — APPROVED** [review]
LGTM 🙇

**Review by bpocrafael — APPROVED** [review]

**Pull request merged**